base-models/src/main/java/com/telecominfraproject/wlan/core/model/utils/DecibelUtils.java

```diff
@@ -19,6 +22,27 @@ public static double getAverageDecibel(int[] decibels)
         return avgDbVal10;
     }
 
+    public static double getAverageDecibel(Collection<Integer> decibels) 
+    {
+    	if (decibels == null || decibels.isEmpty()) { 
+    		return 0.0; 
+    	}
+    	
+    	double aggregation = 0; 
+    	int count=0; 
+    	Iterator<Integer> iter = decibels.iterator(); 
+    	
+    	while(iter.hasNext()) { 
+    		Integer value = iter.next(); 
+    		if(value!=null) { 
+    			aggregation += Math.pow(10, value/10.0); 
+    			count++; 
+    		} 
+    	} 
+    	
+    	double avgDbVal10 = 10 * Math.log10(aggregation/count); 
+    	return avgDbVal10; 
```

Comment:
The High Priests of the Church of the Sonar demand this change:

if(count == 0){
   return 0.0;
} else {
   return 10 * Math.log10(aggregation/count) ;
}

---
https://sonarcloud.io/project/issues?id=Telecominfraproject_wlan-cloud-base2&pullRequest=11&resolved=false&types=BUG